### PR TITLE
New command to update group mappings (#26341)

### PR DESCRIPTION
**Pull request by rullzer**
See https://github.com/owncloud/core/pull/26341

CC: @blizzz @MorrisJobke @nickvergessen

### CODE REVIEW

File: apps/user_ldap/appinfo/register_command.php [hunk 1]
```diff
@@ -53,3 +53,4 @@
 $application->add(new OCA\User_LDAP\Command\CheckUser(
 	$uBackend, $helper, $deletedUsersIndex, $userMapping)
 );
+$application->add(new OCA\User_LDAP\Command\UpdateGroup(new LDAP(), $helper, $dbConnection));
```
Comment (nickvergessen) [inline apps/user_ldap/appinfo/register_command.php]: Please use https://github.com/nextcloud/server/pull/1705

File: apps/user_ldap/lib/Command/UpdateGroup.php [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+	 * ['added' => ['user1', 'user20'], 'removed' => ['user22']]
+	 *
+	 * @param string $groupName name of the group to be checked
+	 * @param array $userList array of user names to be compared. For example: ['user1', 'user44'].
+	 * The list will usually come from the LDAP server and will be compared against the information
+	 * in the DB.
+	 */
+	private function updateGroupMapping($groupName, $userList) {
+		$query = $this->connection->getQueryBuilder();
+		$needToInsert = false;
```
Comment (nickvergessen) [inline apps/user_ldap/lib/Command/UpdateGroup.php]: not needed

File: apps/user_ldap/lib/Command/UpdateGroup.php [hunk 3]
```diff
@@ -0,0 +1,220 @@
...
+		$result = $query->select('*')
+			->from('ldap_group_members')
+			->where($query->expr()->eq('owncloudname', $query->createParameter('group')))
+			->setParameter('group', $groupName)
```
Comment (nickvergessen) [inline apps/user_ldap/lib/Command/UpdateGroup.php]: use `createNamedParameter($groupName)` to save one row for each usage?

File: apps/user_ldap/lib/Command/UpdateGroup.php [hunk 4]
```diff
@@ -0,0 +1,220 @@
...
+		} else {
+			$needToInsert = true;
+		}
+		if ($needToInsert) {
+			$query2 = $this->connection->getQueryBuilder();
+			$query2->insert('ldap_group_members')
+				->setValue('owncloudname', $query2->createParameter('group'))
+				->setValue('owncloudusers', $query2->createParameter('users'))
+				->setParameter('group', $groupName)
+				->setParameter('users', serialize($userList))
```
Comment (nickvergessen) [inline apps/user_ldap/lib/Command/UpdateGroup.php]: please dont use serialize but json_encode/decode instead.
unserialize has security issues

File: apps/user_ldap/lib/Command/UpdateGroup.php [hunk 5]
```diff
@@ -0,0 +1,220 @@
...
+		$this->helper = $helper;
+		parent::__construct();
+	}
+
+	protected function configure() {
+		$this
+			->setName('ldap:update-group')
+			->setDescription('update the specified group membership information stored locally')
+			->addArgument(
+					'groupID',
```
Comment (nickvergessen) [inline apps/user_ldap/lib/Command/UpdateGroup.php]: indentation is off

File: apps/user_ldap/lib/Command/UpdateGroup.php [hunk 6]
```diff
@@ -0,0 +1,220 @@
...
+use \OCA\User_LDAP\Mapping\GroupMapping;
+use \OCP\IDBConnection;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+
+class UpdateGroup extends Command {
+
+	const ERROR_CODE_MISSING_CONF = 1;
+	const ERROR_CODE_MISSING_MAPPING = 2;
+
+	public function __construct(LDAP $ldap, Helper $helper, IDBConnection $connection) {
```
Comment (nickvergessen) [inline apps/user_ldap/lib/Command/UpdateGroup.php]: docs

**Review by nickvergessen — CHANGES_REQUESTED** [review]

File: apps/user_ldap/appinfo/register_command.php (hunk 1 above)
Comment (rullzer) [inline apps/user_ldap/appinfo/register_command.php]: Let me do that for LDAP in a new PR for all commands

**Comment by blizzz** [comment]
:-1: too much bad code

https://github.com/nextcloud/server/issues/1728

**Comment by rullzer** [comment]
@blizzz okidoki then I'll close this.

**Pull request closed without merging**